### PR TITLE
Hydrator Performance Improvements

### PR DESCRIPTION
**Pull request by elfenheart**
- Cancelling timeouts
- Removing model copies on widgets
- disabled debug info for production

### CODE REVIEW

File: cdap-ui/app/directives/metric-picker/metric-picker.js [hunk 1]
```diff
@@ -175,8 +175,8 @@ angular.module(PKG.name + '.commons')
         }
 
         var onBlurHandler = ngModel.$setTouched.bind(ngModel);
-        elem.find('button').on('blur', onBlurHandler);
-        elem.find('input').on('blur', onBlurHandler);
+        var elem2 = elem.find('button').on('blur', onBlurHandler);
```
Comment (elfenheart) [inline cdap-ui/app/directives/metric-picker/metric-picker.js]: Can we rename these elems to something more descriptive?

**Comment by ajainarayanan** [comment]
Bamboo build: http://builds.cask.co/browse/CDAP-DRC3878

**Comment by elfenheart** [comment]
http://builds.cask.co/browse/CDAP-DRC3878 

build pass

**Comment by ajainarayanan** [comment]
- Reviewed the changes with @elfenheart
- Pushed it to a cluster to test it out and it worked fine.
  Corresponding JIRA: https://issues.cask.co/browse/CDAP-5983

Hence merging the same.

**Pull request merged**